homeassistant/components/aten_pe/switch.py

```diff
@@ -53,13 +54,24 @@ async def async_setup_platform(hass, config, async_add_entities, discovery_info=
         await hass.async_add_executor_job(dev.initialize)
         mac = await dev.deviceMAC()
         outlets = dev.outlets()
+        name = await dev.deviceName()
+        model = await dev.modelName()
+        sw_version = await dev.deviceFWversion()
     except AtenPEError as exc:
         _LOGGER.error("Failed to initialize %s:%s: %s", node, serv, str(exc))
         raise PlatformNotReady from exc
 
+    info = {
+        "connections": {(CONNECTION_NETWORK_MAC, mac)},
+        "name": name,
+        "manufacturer": "ATEN",
+        "model": model,
+        "sw_version": sw_version,
+    }
+
```

Comment:
Please import and use `DeviceInfo` class rather than a standard dict.

```suggestion
    info = DeviceInfo(
        connections= {(CONNECTION_NETWORK_MAC, mac)},
        name= name,
        manufacturer= "ATEN",
        model= model,
        sw_version= sw_version,
    )

```

Reply:
Note: don't commit suggestion as it will fail based on formatting :)

Reply:
OK. The documentation at https://developers.home-assistant.io/docs/device_registry_index/ does not cover that aspect.

Reply:
Aha! the documentation needs to be updated...
It can be imported with `from homeassistant.helpers.entity import DeviceInfo`